src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String line = bufferedReader.readLine();
+            int supply = 0;
+            int buy = 0;
+
+            while (line != null) {
+                String[] lineContent = line.split(",");
+                if (lineContent[0].equals("supply")) {
+                    supply += Integer.parseInt(lineContent[1]);
+                } else if (lineContent[0].equals("buy")) {
+                    buy += Integer.parseInt(lineContent[1]);
+                }
+                line = bufferedReader.readLine();
+            }
+
+            int result = supply - buy;
 
+            try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName))) {
```

Comment:
Avoid opening nested connections to files. The BufferedWriter is opened inside the BufferedReader's try block. Refactor this so that each resource is opened in a separate try block and not nested inside each other.